lib/canary/plugs.ex

```diff
@@ -222,26 +225,37 @@ defmodule Canary.Plugs do
   defp fetch_resource(conn, opts) do
     repo = Application.get_env(:canary, :repo)
 
+    id_name = get_resource_id_name(conn, opts)
```

Comment:
Dude, I'm sorry to keep nitpicking on this, but while the opts key should be id_name, I think the variable on this line should just be id, and the function should be get_resource_id rather than get_resource_id_name, since we are really just getting the id. 

That's my fault: I should have been more clear in my last comment. Other than that, I think this looks good!

So, the opts key is id_name, but this variable on line 228 should just be id, and the function should be get_resource_id. 

Let me know if you disagree! :)